keps/sig-storage/1698-generic-ephemeral-volumes/README.md

```diff
@@ -0,0 +1,566 @@
+- [ ] (R) KEP approvers have approved the KEP status as `implementable`
+- [ ] (R) Design details are appropriately documented
+- [ ] (R) Test plan is in place, giving consideration to SIG Architecture and SIG Testing input
+- [X] (R) Graduation criteria is in place
+- [ ] (R) Production readiness review completed
+- [ ] Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+This KEP proposes an alternative mechanism for specifying volumes
+inside a pod spec. Those inline volumes then get converted to normal
+volume claim objects for provisioning, so storage drivers do not need
+to be modified.
+
+The lifecycle of those volumes is ephemeral: they will get deleted
+once the pod terminates.
+
+Because this is expected to be slower than [CSI ephemeral inline
+volumes](https://github.com/kubernetes/enhancements/issues/596), both
+approaches will need to be supported.
+
+
+## Motivation
+
+The current CSI ephemeral inline volume feature has demonstrated that
+ephemeral inline volumes are a useful concept, for example to provide
+additional scratch space for a pod. Several CSI drivers support it
+now, most of them in addition to normal volumes (see
+https://kubernetes-csi.github.io/docs/drivers.html).
+
+However, the original design was intentionally focused on
+light-weight, local volumes. It is not a good fit for volumes provided
+by a more traditional storage system because:
+- The normal API for selecting volume parameters (like size and
+  storage class) is not supported.
+- Integration into storage capacity aware pod scheduling is
+  challenging and would depend on extending the CSI ephemeral inline
+  volume API.
+- CSI drivers need to be adapted and have to take over some of the
+  work normally done by Kubernetes, like tracking of orphaned volumes.
+
+
+### Goals
+
+- Volumes can be specified inside the pod spec ("inline").
+- Volumes are created for specific pods and deleted after the pod
+  terminates ("ephemeral").
+- A normal, unmodified storage driver can be selected via a storage class.
+- The volume will be created using the normal storage provisioning
+  mechanism, without having to modify the driver or its deployment.
+- Storage capacity tracking can be enabled also for such volumes.
```

Comment:
Eventual API alignment with CSI ephemeral inline volumes, as well as current ephemeral in-tree volumes (EmptyDir, etc.), to ensure consistency across ephemeral volume APIs and minimize user confusion.